Packages/com.latios.latiosframework/Mimic/Mecanim/Internal/MecanimInternalUtilities.cs

```diff
@@ -458,11 +459,172 @@ internal static bool Approximately(float a, in float b)
             }
         }
 
+        internal static TransformQvvs GetRootMotionDelta(ref MecanimControllerBlob controllerBlob,
+                                             ref SkeletonClipSetBlob clipSet,
+                                             NativeArray<MecanimParameter> parameters,
+                                             float deltaTime,
+                                             NativeList<TimedMecanimClipInfo> clipWeights,
+                                             DynamicBuffer<TimedMecanimClipInfo> previousFrameClipInfo,
+                                             float totalWeight,
+                                             float weightCullingThreshold)
+        {
+            //Get the current clip deltas
+            var currentRoot = TransformQvvs.identity;
+            for (int i = 0; i < clipWeights.Length; i++)
+            {
+                var clipWeight = clipWeights[i];
+                ref var clip = ref clipSet.clips[clipWeight.mecanimClipIndex];
+                var blendWeight = clipWeight.weight / totalWeight;
+                //Cull clips with negligible weight
+                if (blendWeight < weightCullingThreshold)
+                    continue;
+
+                ref var state = ref controllerBlob.layers[clipWeight.layerIndex].states[clipWeight.stateIndex];
+                var time = state.isLooping ? clip.LoopToClipTime(clipWeight.motionTime) : math.min(clipWeight.motionTime, clip.duration);
+                var stateSpeed = state.speedMultiplierParameterIndex != -1 ?
+                                 parameters[state.speedMultiplierParameterIndex].floatParam * state.speed :
+                                 state.speed;
+                var speedModifiedDeltaTime = deltaTime * stateSpeed;
+                var deltaTransform = TransformQvvs.identity;
+                var hasLooped = state.isLooping && time - deltaTime < 0f;
+
+                //If the clip has looped, get a sample of the end of the clip to incorporate it into the delta
+                if (hasLooped)
+                {
+                    deltaTransform = clip.SampleBone(0, time);
+                    var previousClipSample = clip.SampleBone(0, time - speedModifiedDeltaTime);
+
+                    var sampleEnd = math.select(clip.duration, -clip.duration, stateSpeed < 0f);
+                    var endClipSample = clip.SampleBone(0, sampleEnd);
+
+                    deltaTransform.position += endClipSample.position - previousClipSample.position;
+                    deltaTransform.rotation = math.mul(deltaTransform.rotation, math.mul(math.inverse(endClipSample.rotation), previousClipSample.rotation));
+                }
```

Comment:
Again, not a big deal if you don't want to support this yet, but you don't have scale and stretch factored in here. They use the same math. They function like position, except every addition is multiplication and every subtraction is division.

Reply:
I will transform the other values.  Completely forgot our conversation about QVVS.